test/Translate/array_record_insert.qke

```diff
@@ -0,0 +1,248 @@
+// CHECK:           quake.h %[[VAL_2]] : (!quake.ref) -> ()
+// CHECK:           quake.h %[[VAL_2]] : (!quake.ref) -> ()
+// CHECK:           quake.x {{\[}}%[[VAL_1]]] %[[VAL_2]] : (!quake.ref, !quake.ref) -> ()
+// CHECK:           quake.h %[[VAL_2]] : (!quake.ref) -> ()
+// CHECK:           %[[VAL_3:.*]] = cc.alloca !cc.array<i8 x 2>
+// CHECK:           %[[VAL_4:.*]] = cc.string_literal "array<i8 x 2>" : !cc.ptr<!cc.array<i8 x 14>>
+// CHECK:           %[[VAL_5:.*]] = cc.cast %[[VAL_4]] : (!cc.ptr<!cc.array<i8 x 14>>) -> !cc.ptr<i8>
+// CHECK:           %[[VAL_6:.*]] = arith.constant 2 : i64
+// CHECK:           call @__quantum__rt__array_record_output(%[[VAL_6]], %[[VAL_5]]) : (i64, !cc.ptr<i8>) -> ()
+// CHECK:           %[[VAL_7:.*]] = quake.mz %[[VAL_1]] name "r00000" : (!quake.ref) -> !quake.measure
+// CHECK:           %[[VAL_8:.*]] = quake.discriminate %[[VAL_7]] : (!quake.measure) -> i1
+// CHECK:           %[[VAL_9:.*]] = cc.cast %[[VAL_3]] : (!cc.ptr<!cc.array<i8 x 2>>) -> !cc.ptr<i8>
+// CHECK:           %[[VAL_10:.*]] = cc.cast unsigned %[[VAL_8]] : (i1) -> i8
+// CHECK:           cc.store %[[VAL_10]], %[[VAL_9]] : !cc.ptr<i8>
+// CHECK:           %[[VAL_11:.*]] = quake.mz %[[VAL_2]] name "r00001" : (!quake.ref) -> !quake.measure
+// CHECK:           %[[VAL_12:.*]] = quake.discriminate %[[VAL_11]] : (!quake.measure) -> i1
+// CHECK:           %[[VAL_13:.*]] = cc.compute_ptr %[[VAL_3]][1] : (!cc.ptr<!cc.array<i8 x 2>>) -> !cc.ptr<i8>
+// CHECK:           %[[VAL_14:.*]] = cc.cast unsigned %[[VAL_12]] : (i1) -> i8
+// CHECK:           cc.store %[[VAL_14]], %[[VAL_13]] : !cc.ptr<i8>
+// CHECK:           quake.dealloc %[[VAL_0]] : !quake.veq<2>
+// CHECK:           return
+// CHECK:         }
+
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__multi_vector() attributes {"cudaq-entrypoint", "cudaq-kernel"} {
+// CHECK:           %[[VAL_0:.*]] = quake.alloca !quake.veq<4>
+// CHECK:           %[[VAL_1:.*]] = quake.extract_ref %[[VAL_0]][0] : (!quake.veq<4>) -> !quake.ref
+// CHECK:           quake.h %[[VAL_1]] : (!quake.ref) -> ()
+// CHECK:           %[[VAL_2:.*]] = quake.extract_ref %[[VAL_0]][1] : (!quake.veq<4>) -> !quake.ref
+// CHECK:           quake.x {{\[}}%[[VAL_1]]] %[[VAL_2]] : (!quake.ref, !quake.ref) -> ()
+// CHECK:           %[[VAL_3:.*]] = quake.extract_ref %[[VAL_0]][3] : (!quake.veq<4>) -> !quake.ref
+// CHECK:           quake.h %[[VAL_3]] : (!quake.ref) -> ()
+// CHECK:           %[[VAL_4:.*]] = quake.extract_ref %[[VAL_0]][2] : (!quake.veq<4>) -> !quake.ref
+// CHECK:           quake.h %[[VAL_4]] : (!quake.ref) -> ()
+// CHECK:           quake.x {{\[}}%[[VAL_3]]] %[[VAL_4]] : (!quake.ref, !quake.ref) -> ()
+// CHECK:           quake.h %[[VAL_4]] : (!quake.ref) -> ()
+// CHECK:           %[[VAL_5:.*]] = cc.alloca !cc.array<i8 x 2>
+// CHECK:           %[[VAL_6:.*]] = cc.string_literal "array<i8 x 4>" : !cc.ptr<!cc.array<i8 x 14>>
```

Comment:
I think the Run Spec used something like `"[i8 x 4]"`. We should stick with what was documented there.